src/cliffordsynthesis/encoding/SATEncoder.cpp

```diff
@@ -114,7 +127,38 @@ Result SATEncoder::solve() const {
 void SATEncoder::extractResultsFromModel(Results& res) const {
   auto* const model = lb->getModel();
   tableauEncoder->extractTableauFromModel(res, T, *model);
-  gateEncoder->extractCircuitFromModel(res, *model);
+  auto qc = gateEncoder->extractCircuitFromModel(res, *model);
+  if (config.ignoreRChanges) {
+    const auto start = std::chrono::high_resolution_clock::now();
+    auto       tab   = *config.targetTableau;
+    auto       qcTab = Tableau(qc, 0, std::numeric_limits<std::size_t>::max(),
+                               config.targetTableau->hasDestabilizers());
```

Comment:
If you were to implement the change proposed above, I would guess that this can be skipped completely as the resulting tableau from the circuit is available in the results.